private/bufpkg/bufmodule/bufmodulestore/module_data_store.go

```diff
@@ -263,49 +296,12 @@ func (p *moduleDataStore) getModuleDataForModuleKey(
 	), nil
 }
 
-func (p *moduleDataStore) deleteInvalidModuleData(
-	ctx context.Context,
-	moduleKey bufmodule.ModuleKey,
-	invalidErr error,
-) (retErr error) {
-	p.logDebugModuleKey(
-		moduleKey,
-		"module data store invalid module data",
-		zap.Error(invalidErr),
-	)
-	defer func() {
-		if retErr != nil {
-			// Do not return error, just log. We always returns a file not found error.
-			p.logDebugModuleKey(
-				moduleKey,
-				"module data store could not delete module data",
-				zap.Error(retErr),
-			)
-		}
-		// This will act as if the file is not found.
-		retErr = &fs.PathError{Op: "read", Path: moduleKey.String(), Err: fs.ErrNotExist}
-	}()
-
-	if p.tar {
-		tarPath, err := getModuleDataStoreTarPath(moduleKey)
-		if err != nil {
-			return err
-		}
-		return p.bucket.Delete(ctx, tarPath)
-	}
-	dirPath, err := getModuleDataStoreDirPath(moduleKey)
-	if err != nil {
-		return err
-	}
-	return p.bucket.DeleteAll(ctx, dirPath)
-}
-
 func (p *moduleDataStore) putModuleData(
```

Comment:
I don't see anywhere where invalid data is cleaned up anymore. Perhaps I missed it. What happens if I have invalid  data, say file "a.proto", and the valid data is only the file "b.proto", both written to same cache directory...will "a.proto" not be there anymore? Basically, I don't understand how the deleteInvalidModuleData case is now handled. It appears to only be handled for tar files (`getModuleDataForModuleKey` calls `bucket.Delete`).

Reply:
So the validity of the contents of `v3/modules/<digest_type>/<module_key>` is determined by the presence of a valid `module.yaml` file. As documented at the end of `getModuleDataForModuleKey`, we rely on `module.yaml` to be written last.

So the case you're describing, where there is an invalid `a.proto` but a valid `b.proto`, this occurs because of some interruption to the process while writing `a.proto`. In which case, no `module.yaml` would be written, this would not be considered valid, and then new data is fetched, etc.